IPs/RSKIP170.md

```diff
@@ -0,0 +1,87 @@
+
+The BTC transaction sent to the federation must contain 
+ one output with value 0 and OP_RETURN op code followed by the following data:
+- `52534b54` as prefix indicating the output is for RSK. This values stands for `RSKT` (for RSK  transfer) in ascii and hex encoded. (4 bytes)
+- Protocol version in hexa (1 byte). Currently **01**
+- A valid RSK address (20 bytes)
+- [Optional] A valid BTC refund address in the following format:
+    - Address type (1 byte)
+        - 01 for P2PKH
+        - 02 for P2SH
+    - Hash needed to build refund address (20 bytes)
+        - For P2PKH type address: public key hash 
+        - For P2SH type address: script hash
+
+### Example output
+```
+{
+      "value": 0.00000000,
+      "n": 1,
+      "scriptPubKey": {
+        "asm": "OP_RETURN 52534b5400010e537aad84447a2c2a7590d5f2665ef5cf9b667a014f4c767a2d308eebb3f0f1247f9163c896e0b7d2",
+        "hex": "6a2b52534b54010e537aad84447a2c2a7590d5f2665ef5cf9b667a014f4c767a2d308eebb3f0f1247f9163c896e0b7d2",
+        "type": "nulldata"
+      }
+    }
+```
+Payload included: `52534b54010e537aad84447a2c2a7590d5f2665ef5cf9b667a014f4c767a2d308eebb3f0f1247f9163c896e0b7d2`
+
+- First 4 bytes correspond to the prefix that indicates the output has peg-in instructions for RSK (`RSKT` in hexa): `52534b54` 
+- Next 1 byte that correspond to the version number in hexa: `01`
+- Next 20 bytes indicate rsk destination address: `0e537aad84447a2c2a7590d5f2665ef5cf9b667a`
+- Next 1 byte indicating btc refund address type, in this case P2PKH: `01`
+- Finally 20 bytes indicating the public key hash that will be used to get the btc refund address: `4f4c767a2d308eebb3f0f1247f9163c896e0b7d2`
+
+### New peg-in flow
+
+When a BTC transaction is sent to the Bridge by a federate node, the following steps take place:
+
+1. Blockchain validations as performed in the existing flow (e.g. amount of confirmations, PMT valid)
+2. Parse transaction to get RSK destination address and BTC refund address
+   1. If no OP_RETURN data is present, follow the current flow. Get tx sender to use as BTC refund address and derive RSK destination address from BTC public key.
+   2. If OP_RETURN data is present, get RSK destination address from the payload and the BTC refund address if present. In case there is no BTC refund address included in the payload, get the tx sender.
+
+### Notes
+- There is no limit to the amount of outputs with OP_RETURN the peg-in transaction can have. But there can be only one with `RSKT` prefix indicating peg-in instructions. Transactions with more than one OP_RETURN outputs containing `RSKT` prefix will be rejected, following the existing protocol for refunds.
+- No restriction is added to the maximum length in bytes of the transaction.
+- There is no limit to the amount of outputs the transaction can have, as long as there is no more than one with OP_RETURN op code containing `RSKT` prefix in its payload.
```

Comment:
This note seems duplicated with the first one

Reply:
Actually, this one refers to the total amount of outputs while the first one refers to the amount of outputs with OP_RETURN. I agree it is not clear, I changed it now